packages/amplify_api_rest/android/src/main/kotlin/com/amazonaws/amplify/amplify_api_rest/AmplifyApiRestPlugin.kt

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_api_rest
```

Comment:
It is unconventional to use a `_` in a java package name. I forget if you guys are doing this for other packages?

Reply:
yeah we do that in our other packages as well: 

package com.amazonaws.amplify.amplify_datastore

what is the normal convention? @jamesonwilliams

Reply:
should it have been package.com.amazonaws.amplify.amplifyApiRest instead?

Reply:
The [Java Language Specification (JLS)](https://docs.oracle.com/javase/specs/jls/se6/html/packages.html#7.7 actually allows the organization to define a convention for sub-package names (the part after `com.amazonaws`, which is our Internet domain name in reverse.) So, in a sense, we can use any valid string.

De facto, though, there is a standard ([see discussion](https://stackoverflow.com/questions/49890803/naming-conventions-of-composed-package-names)), which is to use `lower_snake_case` but without any underscores.

So, `com.amazonaws.amplify.amplify_api_rest` might be written better as:
- `com.amazonaws.amplify.api.rest`, or
- `com.amazonaws.amplify.restapi`

Or something like that.

Anyway, given that all of the other packages follow the convention you're currently using, right now, it's probably better to keep things uniform right now.

Reply:
Thanks for the contextual information there!  Okay sounds good.